revise docs (#547)
open-mmlab/mmengine: README.md

## Introduction

MMEngine is a foundational library for training deep learning models based on PyTorch. It provides a solid engineering foundation and frees developers from writing redundant codes on workflows. It serves as the training engine of all OpenMMLab codebases, which support hundreds of algorithms in various research areas. Moreover, MMEngine is also generic to be applied to non-OpenMMLab projects.

Major features:

1. **A universal and powerful runner**:

   - Supports training different tasks with a small amount of code, e.g., ImageNet can be trained with only 80 lines of code (400 lines of the original PyTorch example)
   - Easily compatible with models from popular algorithm libraries such as TIMM, TorchVision, and Detectron2

2. **Open architecture with unified interfaces**:

   - Handles different algorithm tasks with unified APIs, e.g., implement a method and apply it to all compatible models.

>    - Provides a unified abstraction for the upper-level algorithm library, which supports various back-end devices such as Nvidia CUDA, Mac MPS, AMD, MLU, and other devices for model training.

Reviewer: ```suggestion
   - Provides a unified abstraction for upper-level algorithm libraries, which supports various back-end devices such as Nvidia CUDA, Mac MPS, AMD, MLU, and more for model training.
```
